File: test1/app/src/main/java/com/example/test1/MainActivity.java
package com.example.test1;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Button;
import android.view.View;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity {

    private boolean flag = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button button = findViewById(R.id.button);

        final ImageView imageView = findViewById(R.id.imageView);
        imageView.setImageResource(R.drawable.neko1);


        button.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View view) {
                // flagがtrueの時
                if (flag) {

                    imageView.setVisibility(View.INVISIBLE);
                    flag = false;

                } else {

                    imageView.setVisibility(View.VISIBLE);
                    flag = true;

                }
            }

        });


    }
}

File: test2/app/src/main/java/com/example/test2/MainActivity.kt
package com.example.test2

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText

class MainActivity : AppCompatActivity() {

    val EXTRA_MESSAGE: String = "com.example.test2.MESSAGE"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

    }

    fun sendMessage(@Suppress("UNUSED_PARAMETER")view: View) {
        // val intent:として明示的に型を指定する
        val intent: Intent = Intent(this@MainActivity,
            DisplayMessageActivity::class.java)
        val editText: EditText = findViewById(R.id.editText) as EditText
        val message: String = editText.text.toString()
        intent.putExtra(EXTRA_MESSAGE, message)
        startActivity(intent)
    }

}
